search.go

```diff
@@ -79,8 +79,15 @@ func searchHandlerWithProxyMode(logger *zap.Logger, indexer Indexer, proxyMode *
 		serveJSONResponse(r.Context(), w, cacheTime, data)
 
 		if cache != nil {
-			val := cache.Add(r.URL.String(), data)
-			logger.Debug("added to cache request", zap.String("cache.url", r.URL.String()), zap.Int("cache.size", cache.Len()), zap.Bool("cache.eviction", val))
+			switch {
+			case filter.PackageName != "" && !filter.AllVersions:
+				// requests like `/search?package=foo` are not added to the cache
+				// but `/search?package=foo&all=true` is added
```

Comment:
Nit. Mention why we are doing this.

Reply:
Added reasoning about this in this commit: https://github.com/elastic/package-registry/pull/1378/commits/f8cb957aedcf3eb991fe4a2f4e7ad04b263e2be9